grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java

```diff
@@ -245,25 +267,20 @@ public void waitToFinish(){
      * @param queries Queries to be inserted
      */
     void sendQueriesToLoader(Collection<Query> queries){
-        try {
-            blocker.acquire();
```

Comment:
YEEESSSSS!!!!! . . .  What the hell did this actually do????

Reply:
I can't remember exactly but there was a lot of extra logic to make sure we had at most N requests running in parallel, which can be done with a threadpool served by a blocking queue and no extra synchronisation